### PR TITLE
Added in DM-to-distance converter, and added tests for both electron density models.

### PR DESCRIPTION
**Pull request by GrahamDoskoch**
I've added in a function that allows the user to compute the distance to a source given the DM, coordinates and a choice of electron density model (NE2001 or YMW16). I also added some tests, both for zero DM (as a sanity check) and a non-zero DM.

### CODE REVIEW

**Comment by devanshkv** [comment]
Thanks for your PR, you would have to add `pygedm` to the requirements.txt file. Don't forget to add the version as well. This should fix the error you are getting while running automated tests.

File: arts/observation.py [hunk 1]
```diff
@@ -56,3 +60,25 @@ def beamsize(self):
...
+
+        """
+        coords = SkyCoord(ra, dec, frame="icrs")
+        coords = coords.galactic
+        l, b = coords.l, coords.b
+        # convert to galactic coordinates
+
+        dist, tau_sc = pygedm.dm_to_dist(l, b, dm, method=model)
+
+        return dist
```
Comment (KshitijAggarwal) [inline arts/observation.py:84]: Does this return a float or an astropy object?

File: tests/test_observation.py [hunk 2]
```diff
@@ -18,3 +22,19 @@ def test_min_flux():
...
+def test_dm_to_dist():
+    obs = Observation()
+
+    assert obs.dm_to_dist("00h00m00s", "00d00m00s", 0, "ne2001") == 0
+    assert obs.dm_to_dist("00h00m00s", "00d00m00s", 0, "ymw16") == 0
+
+    assert (
+        pytest.approx(obs.dm_to_dist("17h45m40s", "-29d00m28s", 10, "ne2001"), rel=1e-6)
+        == "422.4824905395508 pc"
+    )
```
Comment (KshitijAggarwal) [inline tests/test_observation.py:36]: Not sure if this test would work, as "422.4824905395508 pc" is a string. I think it should be: 422.4824905395508

File: tests/test_observation.py [hunk 3]
```diff
@@ -18,3 +22,19 @@ def test_min_flux():
...
+    assert (
+        pytest.approx(obs.dm_to_dist("17h45m40s", "-29d00m28s", 10, "ymw16"), rel=1e-6)
+        == "517.3521728515625 pc"
+    )
```
Comment (KshitijAggarwal) [inline tests/test_observation.py:40]: Same for this.

File: arts/observation.py (hunk 1 above)
Comment (GrahamDoskoch) [reply arts/observation.py]: Ah, yes, you're right - I've changed it to return a float!

File: tests/test_observation.py (hunk 2 above)
Comment (GrahamDoskoch) [reply tests/test_observation.py:36]: @KshitijAggarwal You're right; I changed the function before pushing and forget to change the test accordingly.

**Comment by GrahamDoskoch** [comment]
I'm seeing that pytest is failing some of the automated tests, and I'm not sure why - I think I have it properly listed in requirements.txt, but I might be wrong.

**Review by devanshkv — APPROVED** [review]

**Comment by KshitijAggarwal** [comment]
It is failing on f2c install. There is only one version of f2c available on pip: https://pypi.org/project/f2c/#history. Do we need some other version?

**Comment by GrahamDoskoch** [comment]
Ah, that was an issue with the version number - what was listed (and what I should have put here) was different from what was displayed when I had installed it; that's fixed. There's an issue with an f2c dependency (FortigateApi) now, and I've tried several ways of listing it without success. I don't recall installing it manually, so I'm at a bit of a loss.

**Comment by KshitijAggarwal** [comment]
I don't think f2c can be installed and used from pip. It is not in the requirements.txt for pygedm. To fix the automated tests you might have to modify the pytest.yml file. [This](https://github.com/FRBs/pygedm/blob/16e34bab61b431ab6ca4d9d83134398dba60dc4a/.github/workflows/python-app.yml#L26) is how pygedm does it. Maybe this can work?

**Comment by devanshkv** [comment]
@KshitijAggarwal thanks that fixed it.

@GrahamDoskoch please run [black](https://github.com/psf/black) and it should be fine after that.

**Review by KshitijAggarwal — APPROVED** [review]
Looks good now.

**Comment by GrahamDoskoch** [comment]
@KshitijAggarwal Thank you so much! It makes sense that that was the issue. @devanshkv I've run it, and the reformatting has been successful.
